SharedSources/MediaDataSource.swift

```diff
@@ -27,7 +28,10 @@ public class MediaDataSourceAndDelegate:NSObject, UICollectionViewDataSource, UI
     }
 
     public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return Int(services.mediaDataSource.numberOfFiles())
+        let numItems = Int(services.mediaDataSource.numberOfFiles())
+        let hasMedia = numItems > 0
+        collectionView.backgroundView?.isHidden = hasMedia
```

Comment:
This code doesn't belong here. A view should never be modified from a model class

Reply:
we should also consider not just hiding the emptyview but consider removing it. It might lead to side effects having this view still in the view hierarchy